header/headertest/testing.go

```diff
@@ -330,9 +330,10 @@ func ExtendedHeaderFromEDS(t *testing.T, height uint64, eds *rsmt2d.ExtendedData
 	gen.ValidatorsHash = valSet.Hash()
 	gen.NextValidatorsHash = valSet.Hash()
 	gen.Height = int64(height)
-
+	blockID := RandBlockID(t)
+	blockID.Hash = gen.Hash()
 	voteSet := types.NewVoteSet(gen.ChainID, gen.Height, 0, tmproto.PrecommitType, valSet)
-	commit, err := MakeCommit(RandBlockID(t), gen.Height, 0, voteSet, vals, time.Now())
+	commit, err := MakeCommit(blockID, gen.Height, 0, voteSet, vals, time.Now())
```

Comment:
this method was not introduced in that version. I've added it only in feature_branch

Reply:
Which version? Yeah, but it will be merged as one big integrate blob service commit, while this code is def unrelated

Reply:
This method is absolutely new and is used only in the feature branch. responding to your next question: "Why I haven't merged in main?" - because then I would need to test this method somehow. But anyway It will be used in tests of the blob service

Reply:
Ah, ok, so do you say that `ExtendedHeaderFromEDS`(it's function, not method btw) is new? I thought it was already in main; ok then.